internal/netsync/manager.go

```diff
@@ -1454,14 +1464,19 @@ func (m *SyncManager) handleHeadersMsg(hmsg *headersMsg) {
 			// Skip the block when it has already been requested or is otherwise
 			// already known.
 			hash := &headerHashes[i]
-			_, isRequestedBlock := m.requestedBlocks[*hash]
-			if isRequestedBlock || chain.HaveBlock(hash) {
+			if m.isRequestedBlock(hash) || chain.HaveBlock(hash) {
 				continue
 			}
 
+			// Stop requesting when the request would exceed the max size of the
+			// map used to track requests.
+			if len(m.requestedBlocks)+1 > maxRequestedBlocks {
```

Comment:
Just highlighting that this is a change in behavior. Previously `limitAdd` would enforce limits onto `m.requestedBlocks` and `peer.requestedBlocks` separately, but with the new code `peer.requestedBlocks` is not being limited.

Reply:
Indeed.  I probably should've called this out in the description, but I changed it because the two maps really need to stay in sync.  Otherwise, peers can be hit with misbehavior if expected entries are not in one map or the other.

It's currently not an issue because it's impossible to hit the limiting cases of `limitAdd` because there is a single sync peer and there are never more outstanding requests than `maxInFlightBlocks = 16` while the limit was set to `maxRequestedBlocks = wire.MaxInvPerMsg =  50000`.